systemtests/src/main/java/com/github/streamshub/systemtests/setup/StrimziOperatorSetup.java

```diff
@@ -0,0 +1,306 @@
+        return strimziResources.stream().filter(file -> file.getName().matches(".*-Crd-.*\\.yaml"))
+            .map(crd -> new CustomResourceDefinitionBuilder(SetupUtils.configFromYaml(crd, CustomResourceDefinition.class))
+                .editMetadata()
+                    .withNamespace(this.deploymentNamespace)
+                .endMetadata()
+                .build())
+            .toArray(CustomResourceDefinition[]::new);
+    }
+
+    private ClusterRole[] getBundleClusterRoles() {
+        return strimziResources.stream().filter(file -> file.getName().matches(".*-ClusterRole-.*\\.yaml"))
+            .map(file -> new ClusterRoleBuilder(SetupUtils.configFromYaml(file, ClusterRole.class))
+                .editMetadata()
+                    .withNamespace(this.deploymentNamespace)
+                .endMetadata()
+                .build())
+            .toArray(ClusterRole[]::new);
+    }
+
+    private RoleBinding[] getBundleRoleBindings() {
+        return strimziResources.stream().filter(file -> file.getName().matches(".*-RoleBinding-.*\\.yaml"))
+            .map(rb -> new RoleBindingBuilder(SetupUtils.configFromYaml(rb, RoleBinding.class))
+                .editMetadata()
+                    .withNamespace(this.deploymentNamespace)
+                .endMetadata()
+                .editFirstSubject()
+                    .withNamespace(this.deploymentNamespace)
+                .endSubject()
+                .build())
+            .toArray(RoleBinding[]::new);
+    }
+
+    private ClusterRoleBinding[] getBundleClusterRoleBindings() {
+        return strimziResources.stream().filter(file -> file.getName().matches(".*-ClusterRoleBinding-.*\\.yaml"))
+            .map(crb -> new ClusterRoleBindingBuilder(SetupUtils.configFromYaml(crb, ClusterRoleBinding.class))
+                .editMetadata()
+                    .withNamespace(this.deploymentNamespace)
+                .endMetadata()
+                .editFirstSubject()
+                    .withNamespace(this.deploymentNamespace)
+                .endSubject()
+                .build())
+            .toArray(ClusterRoleBinding[]::new);
+    }
+
+    private List<File> fetchStrimziYamlFiles() {
+        List<File> yamlFiles = new ArrayList<>();
+        for (String fileUrl : extractYamlFileLinksFromGithubDir()) {
+            String fileName = fileUrl.substring(fileUrl.lastIndexOf('/') + 1).replace(".yaml", "");
+            LOGGER.debug("Fetching Strimzi YAML file: {}", fileName);
+            Path tempFile = null;
+            try {
+                tempFile = Files.createTempFile(fileName + "-tmp_", ".yaml");
+                Files.write(tempFile, SetupUtils.getYamlFileContent(fileUrl).getBytes());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            yamlFiles.add(tempFile.toFile());
+        }
+        return yamlFiles;
+    }
+
+    private List<String> extractYamlFileLinksFromGithubDir() {
```

Comment:
Maybe it's something common that you can use for other operators/components/files as well? IDK if you have something common. There are some things in test-frame as well -> e.g. `TestFrameUtils#getFileFromResourceAsStream` etc.
Maybe it will be useful for you as well?

Reply:
I guess i could use configFromYaml then, thanks for bringing that up.